azure-pipelines.yml

```diff
@@ -0,0 +1,486 @@
+  - name: SkipTests
+    value: true
+  # Set build as stable to remove build number from package names, used for milestone builds
+  - name: IsStable
+    value: false
+  # Set Official Build Id
+  - name: OfficialBuildId
+    value: $(Build.BuildNumber)
+  # Produce Test build for PR and Public builds
+  - ${{ if or(eq(variables['System.TeamProject'], 'public'), in(variables['Build.Reason'], 'PullRequest')) }}:
+    - name: SignType
+      value: test
+  # Set variables only for Official build from internal
+  - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+    - name: SignType
+      value: real
+    # PublishType : nopublishtype used for publishing to dotnetblob feed, dotnet myget and symbol server. 
+    - name: PublishType
+      value: nopublishtype
+
+    # Variable groups
+    - group: DotNet-Blob-Feed
+    - group: DotNet-Symbol-Server-Pats
+    # Used for publishing individual leg assets to azure blobstorage
+    - group: DotNet-DotNetCli-Storage
+    # Used for publishing to myget
+    - group: DotNet-MyGet-Publish
+    # Used for dotnet/versions update
+    - group: DotNet-Versions-Publish
+
+    # BlobFeed update
+    - name: _PublishBlobFeedUrl
+      value: https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json
+    - name: _BlobFeedArgs
+      value: /p:PackagesUrl=$(_PublishBlobFeedUrl)
+        /p:SymbolPackagesUrl=$(_PublishBlobFeedUrl)
+        /p:TransportFeedAccessToken=$(dotnetfeed-storage-access-key-1)
+
+    # Symbol Server update
+    - name: _SymbolServerPath
+      value: https://microsoftpublicsymbols.artifacts.visualstudio.com/DefaultCollection
+    - name: _SymbolServerArgs
+      value: /p:SymbolServerPath=$(_SymbolServerPath)
+         /p:SymbolServerPAT=$(microsoft-symbol-server-pat)
+         /p:SymbolExpirationInDays=365
+
+    # ******** Official values *****
+    - name: _CommonPublishArgs
+      value: /p:AzureAccountName=dotnetcli
+        /p:ContainerName=dotnet
+        /p:AzureAccessToken=$(dotnetcli-storage-key)
+        /p:ChecksumAzureAccountName=dotnetclichecksums
+        /p:ChecksumContainerName=dotnet
+        /p:ChecksumAzureAccessToken=$(dotnetclichecksums-storage-key)
+    # Used for publishing to dotnet myget account
+    - name: MyGetApiKey
+      value: $(dotnet-myget-org-api-key )
+    - name: MyGetFeedUrl
+      value: https://dotnet.myget.org/F/dotnet-core/api/v2/package
+    - name: MyGetSymbolsFeedUrl
+      value: https://dotnet.myget.org/F/dotnet-core/symbols/api/v2/package
+    - name: _NugetFeedArgs
+      value: /p:NuGetFeedUrl=$(MyGetFeedUrl)
+        /p:NuGetSymbolsFeedUrl=$(MyGetSymbolsFeedUrl)
+        /p:NuGetApiKey=$(MyGetApiKey)
+jobs:
+# Platform: Linux 
+- template: /eng/jobs/linux-build.yml
+  parameters:
+    Jobs:
+    # Linux glibc X64 leg
+    - job: 
+      displayName: Linux_x64_glibc
+      agentOS: Linux
+      variables:
+        RunArguments: docker run --privileged --rm
```

Comment:
Another thing that seems like it would help simplify and reduce duplication is to have these Run, Build, and Publish args defined in the template and then define args for the values that are different.  For example the RunArgs are the same except for the image/container name and an optional environment variable.  Encapsulating this in a single spot would help avoid unintentional differences.